### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Sites): constructors for Mayer-Vietoris squares

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we give constructors for Mayer-Vietoris squares, which allows the construction of these squares in the case of topological spaces.

---

- [x] depends on: #14959
- [x] depends on: #14957
- [x] depends on: #14951
- [x] depends on: #14890
- [x] depends on: #14935 
- [x] depends on: #14936

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Sites/MayerVietorisSquare.lean [hunk 1]
```diff
@@ -52,10 +51,28 @@ namespace CategoryTheory
 
 open Limits Opposite
 
+variable {C : Type u} [Category.{v} C]
+  {J : GrothendieckTopology C} [HasWeakSheafify J (Type v)]
+
+lemma Sheaf.isPullback_square_op_map_yoneda_presheafToSheaf_yoneda_iff
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Sites/MayerVietorisSquare.lean:57]: Should this even be a simp lemma?

**Review by jcommelin — COMMENTED** [review]
LGTM

bors d+

File: Mathlib/CategoryTheory/Sites/MayerVietorisSquare.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Sites/MayerVietorisSquare.lean:58]: Thanks for the suggestion!

**Comment by joelriou** [comment]
Thanks for the review!

bors merge

**Pull request closed without merging**